### PR TITLE
Fix counts endianness

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**
https://github.com/PennyLaneAI/catalyst/issues/394

**Description of the Change:**
Switch the endianness of counts to match PennyLane.

**Benefits:**
Follow the current convention.

### CODE REVIEW

**Comment by rmoyard** [comment]
Closes https://github.com/PennyLaneAI/catalyst/issues/394

File: runtime/lib/backend/lightning/lightning_dynamic/LightningSimulator.cpp [hunk 1]
```diff
@@ -374,9 +374,9 @@ void LightningSimulator::Counts(DataView<double, 1> &eigvals, DataView<int64_t,
     // into a bitstring.
     for (size_t shot = 0; shot < shots; shot++) {
         std::bitset<CHAR_BIT * sizeof(double)> basisState;
-        size_t idx = 0;
+        size_t idx = numQubits - 1;
         for (size_t wire = 0; wire < numQubits; wire++) {
-            basisState[idx++] = li_samples[shot * numQubits + wire];
+            basisState[idx--] = li_samples[shot * numQubits + wire];
```
Comment (maliasadi) [inline runtime/lib/backend/lightning/lightning_dynamic/LightningSimulator.cpp:379]: Would you also fix this for LightningKokkos?

**Review by maliasadi — COMMENTED** [review]
Amazing! :100:

File: runtime/lib/backend/lightning/lightning_dynamic/LightningSimulator.cpp (hunk 1 above)
Comment (rmoyard) [reply runtime/lib/backend/lightning/lightning_dynamic/LightningSimulator.cpp:379]: Yes done

File: runtime/lib/backend/lightning/lightning_dynamic/LightningSimulator.cpp [hunk 2]
```diff
@@ -374,9 +374,9 @@ void LightningSimulator::Counts(DataView<double, 1> &eigvals, DataView<int64_t,
     // into a bitstring.
     for (size_t shot = 0; shot < shots; shot++) {
         std::bitset<CHAR_BIT * sizeof(double)> basisState;
-        size_t idx = 0;
+        size_t idx = numQubits;
         for (size_t wire = 0; wire < numQubits; wire++) {
-            basisState[idx++] = li_samples[shot * numQubits + wire];
+            basisState[--idx] = li_samples[shot * numQubits + wire];
```
Comment (dime10) [inline runtime/lib/backend/lightning/lightning_dynamic/LightningSimulator.cpp:379]: Sneaky 😄

**Review by dime10 — APPROVED** [review]

**Pull request merged**